### PR TITLE
Docs enforce autoauth token

### PR DESCRIPTION
**Pull request by catsby**
This PR documents the feature added in #8101 , where Vault Agent can be configured to ignore a token in the request and force the use of the auto-auth token. 

I also renamed the internal variable, I felt the prefix `Force` better conveyed the purpose than the suffix `Enforce`

### CODE REVIEW

File: website/pages/docs/agent/caching/index.mdx [hunk 1]
```diff
@@ -49,6 +49,14 @@ configuration will be overridden if the request already has a token attached,
 in which case, the token present in the request will be used to forward the
 request to the Vault server.
 
+## Forcing Auto-Auth Token
+
+Vault Agent can be configured to force the use of the auto-auth token by using
+the value `force` for the `use_auto_auth_token` option. This configuration
+overrides the default behavior described above in [Using Auth-Auth
+Token](/docs/agent/caching#using-auto-auth-token), and instead ignores any
+existing Vault token in the request and instead uses the auth-auth token.
```
Comment (kalafut) [inline website/pages/docs/agent/caching/index.mdx]:
```suggestion
existing Vault token in the request and instead uses the auto-auth token. 
```

File: website/pages/docs/agent/caching/index.mdx [hunk 2]
```diff
@@ -49,6 +49,14 @@ configuration will be overridden if the request already has a token attached,
...
 request to the Vault server.
 
+## Forcing Auto-Auth Token
+
+Vault Agent can be configured to force the use of the auto-auth token by using
+the value `force` for the `use_auto_auth_token` option. This configuration
+overrides the default behavior described above in [Using Auth-Auth
+Token](/docs/agent/caching#using-auto-auth-token), and instead ignores any
+existing Vault token in the request and instead uses the auth-auth token. 
+
```
Comment (kalafut) [inline website/pages/docs/agent/caching/index.mdx]: What is the behavior is `force` is set, there is no auto-auth token, but the request does contain a token?

Comment (catsby) [reply website/pages/docs/agent/caching/index.mdx]: This "can't" happen; if the configuration has `use_auto_auth_token` set to `true` or `force`, and no valid `auto_auth` block is defined, then Vault Agent will exit with status code `1` at startup

Comment (kalafut) [reply website/pages/docs/agent/caching/index.mdx]: 👍

File: website/pages/docs/agent/caching/index.mdx [hunk 3]
```diff
@@ -160,11 +168,18 @@ $ curl \
...
-- `use_auto_auth_token (bool: false)` - If set, the requests made to agent
+- `use_auto_auth_token (bool/string: false)` - If set, the requests made to agent
   without a Vault token will be forwarded to the Vault server with the
   auto-auth token attached. If the requests already bear a token, this
   configuration will be overridden and the token in the request will be used to
-  forward the request to the Vault server.
+  forward the request to the Vault server. If set to `"force"` Agent will use the
+  auto-auth token, overwriting the attached Vault token if set. 
+
+-> **Note:** Vault Agent does not require configuring auto-auth in order to use the
```
Comment (kalafut) [inline website/pages/docs/agent/caching/index.mdx]: Should this be removed, since `force` can be using w/o auto-auth?

Comment (catsby) [reply website/pages/docs/agent/caching/index.mdx]: Yes! Removed in a65556b6e

**Pull request merged**